### PR TITLE
Add Android client for Oak Functions

### PR DESCRIPTION
**Pull request by ipetr0v**
This change adds an example Android client for Oak Functions.
The default functionality of the current client is to connect to a localhost Weather Service and request temperature.

### CODE REVIEW

File: oak_functions/client/android/src/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    backgroundHandlerThread.start();
+    backgroundHandler = new Handler(backgroundHandlerThread.getLooper());
+
+    Button invokeButton = findViewById(R.id.invokeButton);
+    invokeButton.setOnClickListener(v -> onClick());
+
+    // Set default URI of a Cloud Run application:
+    // https://pantheon.corp.google.com/run/detail/europe-west2/oak-functions-weather-lookup/metrics?project=oak-ci
+    EditText uriInput = findViewById(R.id.uriInput);
+    uriInput.setText("oak-functions-weather-lookup-62sa4xcfia-nw.a.run.app:443");
```
Comment (tiziano88) [inline oak_functions/client/android/src/MainActivity.java:64]: Please use the API gateway URL.

cc @conradgrobler

Comment (ipetr0v) [reply oak_functions/client/android/src/MainActivity.java]: Currently in this PR making it connect to localhost

File: oak_functions/client/android/com/google/oak/functions/android/client/res/layout/activity_main.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <!-- Result. -->
+    <TextView
+        android:id="@+id/resultTextView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:textSize="32dp"
+        android:text="@string/empty"/>
+
+</LinearLayout>
```
Comment (conradgrobler) [inline oak_functions/client/android/com/google/oak/functions/android/client/res/layout/activity_main.xml:45]: nit: resource files are missing line breaks at the end.

File: oak_functions/client/android/com/google/oak/functions/android/client/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    backgroundHandlerThread.start();
+    backgroundHandler = new Handler(backgroundHandlerThread.getLooper());
+
+    Button invokeButton = findViewById(R.id.invokeButton);
+    invokeButton.setOnClickListener(v -> onClick());
+
+    // Set default URI of the Oak Functions application.
+    EditText uriInput = findViewById(R.id.uriInput);
+    // 10.0.2.2 is routed to the host machine by the Android emulator.
+    uriInput.setText("10.0.2.2:8080");
```
Comment (conradgrobler) [inline oak_functions/client/android/com/google/oak/functions/android/client/MainActivity.java:67]: Perhaps add a todo to point this to the API Gateway URL once the deployment works and add API key support.

File: oak_functions/client/android/com/google/oak/functions/android/client/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    EditText requestInput = findViewById(R.id.requestInput);
+    byte[] requestBody = requestInput.getText().toString().getBytes();
+    Request request = Request.newBuilder().setBody(ByteString.copyFrom(requestBody)).build();
+
+    TextView resultTextView = findViewById(R.id.resultTextView);
+    try {
+      AttestationClient client = new AttestationClient(uri);
+      Response response = client.send(request);
+      ByteString responseBody = response.getBody().substring(0, (int)response.getLength());
```
Comment (conradgrobler) [inline oak_functions/client/android/com/google/oak/functions/android/client/MainActivity.java:93]: This should probably check the `StatusCode` on the response and display an error message if it is not `SUCCESS`

**Review by conradgrobler — APPROVED** [review]

File: oak_functions/client/android/com/google/oak/functions/android/client/MainActivity.java (hunk 4 above)
Comment (ipetr0v) [reply oak_functions/client/android/com/google/oak/functions/android/client/MainActivity.java:103]: Done

**Pull request merged**